hist/histv7/inc/ROOT/RAxis.hxx

```diff
@@ -59,7 +59,15 @@ protected:
    ///\}
 
    /// Default construct a RAxisBase (for use by derived classes for I/O)
-   RAxisBase() = default;
+   RAxisBase() noexcept = default;
+
+   /// Virtual destructor needed in this inheritance-based design
+   virtual ~RAxisBase() = default;
```

Comment:
It is the only such function, for now (but that will change in #4974).

However, I believe outlining the destructor will currently break the build of the tests due to [ROOT-10554](https://sft.its.cern.ch/jira/browse/ROOT-10554).

Maybe I should fix this issue in this MR as well, after all...

Reply:
Fixed in latest version of the PR. Also, I added tests for GetBinIndexForLowEdge, which uncovered some issues that I also fixed. You can check the last three commits for a summary of these updates, the rest are unchanged (only rebased on current master in hope of fixing jenkins weirdness).

Reply:
Oh, by the way, since I started touching this... I find it somewhat strange that `GetBinIndexForLowEdge` is only defined for `RAxisEquidistant`, as opposed to being a virtual method of `RAxisBase` that `RAxisIrregular` would implement as well. Should I change that?

Reply:
Yes! But doing that in a subsequent PR sounds okay to me?